crates/nu-cli/src/completions/file_completions.rs

```diff
@@ -157,3 +151,52 @@ pub fn matches(partial: &str, from: &str) -> bool {
     from.to_ascii_lowercase()
         .starts_with(&partial.to_ascii_lowercase())
 }
+
+// escape paths that contains some special characters
+pub fn escape_path_str(path: String) -> String {
+    let mut path = path;
+
+    // List of special characters that need to be escaped
+    let special_characters = b"\\\'\"";
+    let replacements = [b"\\\\", b"\\\'", b"\\\""];
+
+    // Check if path needs to be escaped
+    let needs_escape = path.bytes().fold(false, |acc, x| {
+        acc
+        || x == b'\\' // 0x5c
+        || x == b'`' // 0x60
+        || x == b' '
+        || x == b'\''
+    });
+
+    if needs_escape {
+        // Worst case, escape everything (+2 double quotes)
+        let mut result: Vec<u8> = vec![b'\"'];
```

Comment:
Looks like the initial capacity allocation got lost along the way. Please either add it back or remove the comment about it.